flytekit/core/type_engine.py

```diff
@@ -614,10 +624,13 @@ def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type:
     @lru_cache(typed=True)
     def guess_python_type(self, literal_type: LiteralType) -> Type[T]:  # type: ignore
         if literal_type.simple == SimpleType.STRUCT:
-            if literal_type.metadata is not None and DEFINITIONS in literal_type.metadata:
-                schema_name = literal_type.metadata["$ref"].split("/")[-1]
-                return convert_json_schema_to_python_class(literal_type.metadata[DEFINITIONS], schema_name)
-
+            if literal_type.metadata is not None:
+                if DEFINITIONS in literal_type.metadata:
+                    schema_name = literal_type.metadata["$ref"].split("/")[-1]
+                    return convert_json_schema_to_python_class(literal_type.metadata[DEFINITIONS], schema_name)
```

Comment:
could we create two functions for this? 
1. convert_mashumaro_json_schema_to_python_class
2. convert_marshmallow_json_schema_to_python_class